### ISSUE TITLE
Support with other datasets

### ISSUE THREAD

**Issue by ShahzilGul** [ValerianRey/fed_iot_guard#27]
I cannot run this code with other datasets to find compatibility with large datasets. I have shortened the dataset and made various changes in it but still, it is showing me the same error as below.
![WhatsApp Image 2022-11-19 at 8 21 00 PM](https://user-images.githubusercontent.com/107142628/202918326-89f275ca-ffbb-45f1-b53d-82aa2e6971c0.jpeg)

**Comment by ValerianRey** [comment]
According to the error, you're using a binary cross-entropy loss function with inputs that are not between 0 and 1. Is your dataset made for binary classification? Otherwise you should change the loss function and the model.

**Comment by ShahzilGul** [comment]
The dataset I am working on can be used for both binary and multi-class classification. Thank you for the recommendation I will work on it now.

**Comment by akramIOT** [comment]
During the  Training, the  classifier uses each of the N points in its training set to compute the cross-entropy loss. Since these N points could not be distributed  as a part of Binary classification Model in your scenario, this error is  thrown.
